### PR TITLE
Feature scrm 23 bulk action

### PR DESCRIPTION
**Pull request by gymad**
bulk actions on multiple selected records on list-views

## Description

This PR contains Bulk Action button on list views and restyled Record Selection checkbox
(to see all changes a rebuild group js files needed)

## Motivation and Context

SCRM-23 US 5.1 - Bulk Actions
SCRM-24 US 5.2 - Record Selection

## How To Test This

Select any records on any module's list view and use Bulk Action dropdown functions

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [ ] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [ ] My change requires a change to the documentation.
- [ ] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

### CODE REVIEW

**Comment by daniel-samson** [comment]
# Peer Review

Overall good job. Please review the following issues.

**Issues**
- As a User, when I click on the "Bulk Action" Text. I would expect to see the dropdown. In the same way that the Quick Create Button works. I should be able to click anywhere on the button including the caret.
- As a user, When I click the select the checkbox dropdown and click on "Select All". I would expect the check box in to be checked. When I then select the check box it should then un-check all items. See image below

select_all_checkbox

- In the style.css for the SuiteP theme, you have **"text-transform: uppercase"**. You need to remove this and change the translation text to be upper case. Setting the style to uppercase breaks language translation for languages like Portuguese. See _themes/SuiteP/css/style.css_ line: _11219_

**Comment by samus-aran** [comment]
Additional to the notes made by Daniel:

- The design doesn't completely match the design supplied.  See the images attached.
- A few UI issues need to be resolved to completeness

bulk_actions2
bulk_actions

File: jssource/src_files/include/javascript/sugar_3.js [hunk 1]
```diff
@@ -2468,6 +2468,13 @@ sugarListView.prototype.check_entire_list = function(form, field, value, list_co
 	document.MassUpdate.select_entire_list.value = 1;
 	sugarListView.update_count(list_count, false);
     sugarListView.prototype.toggleSelected();
+	$('input[name="massall"]').prop('checked', true);
+	$('.glyphicon.bootstrap-checkbox').each(function (i, e) {
+		if(typeof refreshBootstrapCheckbox !== 'undefined') {
```
Comment (samus-aran) [inline jssource/src_files/include/javascript/sugar_3.js]: Why are we using renames related to the library?  Shouldn't it be less specific names i.e. refreshCheckbox?

File: themes/SuiteP/css/style.css [hunk 2]
```diff
@@ -11205,4 +11205,85 @@ div#content div#pagecontent form#EditView div.edit.view table tbody tr td input[
         height: initial;
         padding: 0px 0px 15px 0px !important;
     }
+}
+
+#select_actions_disabled_top > a > label.hidden-mobile,
```
Comment (samus-aran) [inline themes/SuiteP/css/style.css]: Again, too long.  Refactor elements to have specific classes to refer to rather than having to many child selectors

**Pull request merged**